client/main_test.go

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
```

Comment:
This is taken from `goleak` [documentation](https://pkg.go.dev/go.uber.org/goleak#readme-quick-start) on how to test for leaks in an entire package, instead of adding a check to every single test. 

This will technically lead to less clear error messages where someone will have to figure out which test is causing the leak in a given package, but I think that's the much better option over adding a check to every single test in the repo.

Reply:
I think you can rename from `main_test.go` to `goleak_test.go` and `TestMain` to `TestGoLeaks` to explain this testing a bit more.

Reply:
the function name must be TestMain, is the standard name understood by go test toolchain. Putting VerifyTestMain in there makes it apply to all tests in the package, which won't happen if you name it  TestGoLeaks.

fwit in Jaeger I recommended naming the files `package_test.go`, but only adding them to packages with multiple test files - if there's only one test file then TestMain can go directly into it.

Reply:
Another thing we did in Jaeger is add a linter script that flags all packages that are missing goleak. Right now it's info-only, but once we fix all packages (<30 remaining) we will make it fail the CI.

https://github.com/jaegertracing/jaeger/issues/5006 for reference

Reply:
Thanks I learned something!

Reply:
I like @yurishkuro suggestion for `package_test.go`. I would still add always this file even only one test, just to remove the possible problem if we add a second test file in the future.

Reply:
I've renamed to `package_test.go` 👍